src/components/bootcamp-interns/MultipleRowsDeleteModal.tsx

```diff
@@ -0,0 +1,57 @@
+import router from 'next/router'
+import { Dialog, DialogTitle, DialogActions, Button, Modal } from '@mui/material'
+
+import { routes } from 'common/routes'
+import { axios } from 'common/api-client'
+import { endpoints } from 'common/api-endpoints'
+import { NotificationStore } from 'stores/bootcamp-interns/NotificationStore'
+
+type Props = {
+  modalProps: {
+    deleteData: string[]
+    setDeleteOpen: React.Dispatch<React.SetStateAction<boolean>>
+    deleteOpen: boolean
+  }
+}
+
+export default function MultipleRowsDeleteModal({
+  modalProps: { deleteData, setDeleteOpen, deleteOpen },
+}: Props) {
+  const { setNotificationMessage, showNotification } = NotificationStore
+  const hasSelectedMultipleRows = deleteData?.length > 1
+  console.log(deleteData)
+
+  const handleDelete = async () => {
+    if (hasSelectedMultipleRows) {
+      deleteData.map(async (id: string) => {
+        await axios.delete(`${endpoints.bootcampIntern.listBootcampIntern.url}/${id}`)
```

Comment:
How do you delete multiple records by providing one `id` to the endpoint?

Is that a typo or unfinished functionality?

Reply:
Actually it works to be honest and I know that it's not the best way , and the best way to do that is to have a deleteMany api functionality to use.

Reply:
It works because you send as many requests as the selected rows - one request per selected row.

Here comes the problem, if you select 20 records and send 20 requests the first one to receive back a response will trigger  `router.push` and will lead you to another page without waiting for the other requests to complete.

In a simple delete request that can be alright, but in a more complex scenario that might become troublesome as the responses wont be utilized